test/PushNotificationTests/APITests.cpp

```diff
@@ -0,0 +1,184 @@
+            VERIFY_IS_TRUE(launchResult);
+
+            WaitForEvent(event, m_failed);
+        }
+
+        TEST_METHOD(ChannelRequestUsingRemoteId)
+        {
+            wil::unique_event event = CreateTestEvent(c_testProtocolScheme_Packaged);
+
+            // This is associated protocol for the MSIX installed app for launch.
+            // Use the ://path to define the component you want to test.
+            Uri launchUri{ c_testProtocolScheme_Packaged + L"://ChannelRequestUsingRemoteId" };
+            auto launchResult = Launcher::LaunchUriAsync(launchUri).get();
+            VERIFY_IS_TRUE(launchResult);
+
+            WaitForEvent(event, m_failed);
+        }
+
+        TEST_METHOD(MultipleChannelRequestUsingSameRemoteId)
+        {
+            wil::unique_event event = CreateTestEvent(c_testProtocolScheme_Packaged);
+
+            // This is associated protocol for the MSIX installed app for launch.
+            // Use the ://path to define the component you want to test.
+            Uri launchUri{ c_testProtocolScheme_Packaged + L"://MultipleChannelRequestUsingSameRemoteId" };
+            auto launchResult = Launcher::LaunchUriAsync(launchUri).get();
+            VERIFY_IS_TRUE(launchResult);
+
+            WaitForEvent(event, m_failed);
+        }
+
+        TEST_METHOD(MultipleChannelRequestUsingMultipleRemoteId)
+        {
+            wil::unique_event event = CreateTestEvent(c_testProtocolScheme_Packaged);
+
+            // This is associated protocol for the MSIX installed app for launch.
+            // Use the ://path to define the component you want to test.
+            Uri launchUri{ c_testProtocolScheme_Packaged + L"://MultipleChannelRequestUsingMultipleRemoteId" };
+            auto launchResult = Launcher::LaunchUriAsync(launchUri).get();
+            VERIFY_IS_TRUE(launchResult);
+
+            WaitForEvent(event, m_failed);
+        }
+
+        TEST_METHOD(ThreeChannelRequestUsingSameRemoteId)
+        {
+            wil::unique_event event = CreateTestEvent(c_testProtocolScheme_Packaged);
+
+            // This is associated protocol for the MSIX installed app for launch.
+            // Use the ://path to define the component you want to test.
+            Uri launchUri{ c_testProtocolScheme_Packaged + L"://ThreeChannelRequestUsingSameRemoteId" };
+            auto launchResult = Launcher::LaunchUriAsync(launchUri).get();
+            VERIFY_IS_TRUE(launchResult);
+
+            WaitForEvent(event, m_failed);
+        }
+
+        TEST_METHOD(BackgroundActivation)
+        {
+            wil::unique_event event = CreateTestEvent(c_testProtocolScheme_Packaged);
+            auto LocalBackgroundTask = winrt::create_instance<winrt::Windows::ApplicationModel::Background::IBackgroundTask>(c_comServerId, CLSCTX_ALL);
+            auto mockBackgroundTaskInstance = winrt::make<MockBackgroundTaskInstance>();
+            LocalBackgroundTask.Run(mockBackgroundTaskInstance);
+            WaitForEvent(event, m_failed);
```

Comment:
When does this failed event fire? Why is it passed and what is it's purpose?

Reply:
I guess protocolevent is fired for success and failedevent is fired for failure? And this method basically verifies failure or success?

Reply:
Yes. WaitForEvent will wait for the named event and if it has not either heard back success/failure for 10 seconds, it will fail.